File: payments/search/python/payments_by_external_reference.py
# coding: UTF-8

import os, sys
import mercadopago

import json

def index(req, **kwargs):
    mp = mercadopago.MP("CLIENT_ID", "CLIENT_SECRET")
    
    filters = {
        "external_reference": "ext1425"
    }

    searchResult = mp.search_payment(filters)

    return json.dumps(searchResult, indent=4)

File: payments/search/nodejs/payments_by_external_reference.js
var MP = require ("mercadopago");

var mp = new MP ("CLIENT_ID", "CLIENT_SECRET");

var filters = {
	"external_reference": "ext1425"
};

mp.searchPayment (filters, function (err, data){
	if (err) {
		console.log (err);
	} else {
		console.log (JSON.stringify (data, null, 4));
	}
});